core/core-customization.el

```diff
@@ -91,8 +92,9 @@ NOTE: Variables defined with a group listed in
       ,(format "%s\n\nTYPE: %s\n" doc type)
       :type ,type
       :group group)
-     (when (not (null ',safe))
-       (put ',symbol 'safe-local-variable ,safe))
+     (pcase ,safe
+       ('t (put ',symbol 'safe-local-variable (lambda (val) (validate-value val ,type t))))
```

Comment:
Note that here it's matching the symbol `'t`, not to be confused with the catch-all case `t`.